src/engine/naive_engine.cc

```diff
@@ -55,20 +69,44 @@ class NaiveEngine final : public Engine {
     opr->opr_name = opr_name;
     return opr;
   }
+
   void DeleteOperator(OprHandle op) override {
     NaiveOpr *opr = op->Cast<NaiveOpr>();
     delete opr;
   }
+
   void Push(OprHandle op, Context exec_ctx, int priority = 0, bool profiling = false) override {
+    Profiler *profiler = Profiler::Get();
     NaiveOpr *opr = op->Cast<NaiveOpr>();
-    this->PushAsync(opr->fn,
-                    exec_ctx,
-                    opr->const_vars,
-                    opr->mutable_vars,
-                    opr->prop,
-                    priority,
-                    PROFILER_MESSAGE(opr->opr_name));
+    opr->profiling = profiling && (profiler->GetMode() == Profiler::kOnlySymbolic);
+    this->PushAsync([&](RunContext ctx, CallbackOnComplete on_complete) {
+#if MXNET_USE_PROFILER
+        if (opr->profiling) {
+          opr->opr_stat = Profiler::Get()->AddOprStat(exec_ctx.dev_type, exec_ctx.dev_id);
+          uint64_t id = std::hash<std::thread::id>()(std::this_thread::get_id());
+          opr->opr_stat->thread_id = id;
+          strncpy(opr->opr_stat->opr_name,
+            opr->opr_name,
+            sizeof(opr->opr_stat->opr_name) - 1);
+          CallbackOnComplete callback = CreateCallback(NaiveEngine::OnComplete, opr);
```

Comment:
@piiswrong  the param in original on_complete is nullptr, it will be called `when no using profiler.`  
this is used when mode="symbolic", if we set in in sync, then is indeed do mode="all" which is not we want.